python/langsmith/run_helpers.py

```diff
@@ -178,13 +219,22 @@ def _setup_run(
     outer_project = _PROJECT_NAME.get()
     langsmith_extra = langsmith_extra or LangSmithExtra()
     parent_run_ = langsmith_extra.get("run_tree") or get_run_tree_context()
+    project_cv = _PROJECT_NAME.get()
     selected_project = (
-        _PROJECT_NAME.get()  # From parent trace
+        project_cv  # From parent trace
         or langsmith_extra.get("project_name")  # at invocation time
         or container_input["project_name"]  # at decorator time
         or utils.get_tracer_project()  # default
     )
-    if not parent_run_ and not utils.tracing_is_enabled():
+    reference_example_id = langsmith_extra.get("reference_example_id")
+    id_ = langsmith_extra.get("run_id")
+    if (
+        not project_cv
```

Comment:
Still trace if you're manually providing run ID, example, project, etc. via the context var